### PR TITLE
Ns/subtaskbutton

### PR DESCRIPTION
**Pull request by nschickle**
Okay, I got rid of the issues and I'll do another PR in a short bit for the Oauth. I tried merging them manually and must've messed up a bit. Oauth stuff should be nearly complete.

### CODE REVIEW

File: src/components/mainLanding.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+interface MainLandProps {
+    showProjectLanding: () => any;
+}
+
+export class MainLanding extends React.Component<MainLandProps, { error: any, isLoaded: boolean, showProjectLanding: () => any}> {
+    private userInfo = new UserInfo(null, null);
+    constructor(props: MainLandProps) {
+        super(props);
+
+        this.state = {
```
Comment (theroyalblah) [inline src/components/mainLanding.tsx:73]: Shouldn't need error/isLoaded right?

File: src/components/signInButton.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    constructor(props: SignInButtonProps) {
+        super(props);
+
+        
+    }
+    public render() {
+                return (
+                        <Container style={styles.button}>
+                            <GoogleLogin
+                            clientId="528310070004-u0clc84o9iktpqi6tjujqe9pq9f6ns2n.apps.googleusercontent.com"
```
Comment (theroyalblah) [inline src/components/signInButton.tsx:46]: the clientId likely is not meant to be seen by the outside world. We should probably not have it stored on GitHub.

File: src/components/subTaskButton.tsx [hunk 3]
```diff
@@ -107,9 +107,8 @@ export class SubTaskButton extends React.Component<SubTaskButtonProps>{
 
     // If the title is too long, we should shorten it to fit the space we have.
     checkNameLength = () => {
-        const maxNameLength = 24;
-        if (this.name.length > maxNameLength) {
-            this.displayedName = this.name.substring(0, maxNameLength);
+        if (this.name.length > 32) {
```
[the remark this answers is not in the document]
Comment (nschickle) [reply src/components/subTaskButton.tsx:110]: Nope, this was something I accidentally left behind in testing, meant to add it back. Thank you!

File: src/components/signInButton.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    email: string;
+}
+
+const testOwner: IUser = { id: null, name: null, email: null };
+
+
+const responseGoogle = (response: any) => {
+    console.log(response);
+    console.log(response.profileObj.name);
+    testOwner.name = response.profileObj.name;
```
[the remark this answers is not in the document]
Comment (nschickle) [reply src/components/signInButton.tsx:27]: I'm not entirely sure how to do that, do you think you may have an idea on how to help?

File: src/components/signInButton.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    id: number;
+    name: string;
+    email: string;
+}
+
+const testOwner: IUser = { id: null, name: null, email: null };
+
+
+const responseGoogle = (response: any) => {
+    console.log(response);
```
[the remark this answers is not in the document]
Comment (nschickle) [reply src/components/signInButton.tsx:25]: Done.

File: src/components/mainLanding.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        };
+    }
+
+    render() {
+        return(
+            <Container fluid style={styles.pageContainer}>
+                <Row style={styles.titleIntroduction}>
+                    <Col style={styles.logo}>
+                        <img
+                            src={require("../img/logo.png")}
```
[the remark this answers is not in the document]
Comment (nschickle) [reply src/components/mainLanding.tsx:86]: Will do!

File: dist/components/newSubTaskButton.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
+"use strict";
```
[the remark this answers is not in the document]
Comment (nschickle) [reply dist/components/newSubTaskButton.js:1]: Yea no problem!

File: src/components/signInButton.tsx (hunk 2 above)
Comment (nschickle) [reply src/components/signInButton.tsx:46]: Also unsure of how to do this..

File: src/components/signInButton.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    public render() {
+                return (
+                        <Container style={styles.button}>
+                            <GoogleLogin
+                            clientId="528310070004-u0clc84o9iktpqi6tjujqe9pq9f6ns2n.apps.googleusercontent.com"
+                            buttonText="Sign In With Google and Launch!"
+                            onSuccess={responseGoogle}
+                            onFailure={responseGoogle}
+                            cookiePolicy={'single_host_origin'}
+                            isSignedIn={true}
```
Comment (JustinLaGree) [inline src/components/signInButton.tsx:49]: Why is this always true?
https://github.com/react-native-community/google-signin

Comment (nschickle) [reply src/components/signInButton.tsx]: No longer always true.

**Pull request merged**